oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/LogoutHandler.java

```diff
@@ -92,170 +92,147 @@ public APIGatewayProxyResponseEvent handleRequest(
-        if (sessionFromSessionCookie.isPresent()) {
-            return segmentedFunctionCall(
-                    "processLogoutRequest",
-                    () -> processLogoutRequest(sessionFromSessionCookie.get(), input, state));
-        } else {
-            return segmentedFunctionCall(
-                    "generateDefaultLogoutResponse",
-                    () ->
-                            logoutService.generateDefaultLogoutResponse(
-                                    state, input, Optional.empty(), Optional.empty()));
-        }
-    }
-
-    private APIGatewayProxyResponseEvent processLogoutRequest(
-            Session session, APIGatewayProxyRequestEvent input, Optional<String> state) throws ParseException {
-
-        CookieHelper.SessionCookieIds sessionCookieIds =
-                cookieHelper.parseSessionCookie(input.getHeaders()).orElseThrow();
-
-        attachSessionIdToLogs(session);
-        attachLogFieldToLogs(CLIENT_SESSION_ID, sessionCookieIds.getClientSessionId());
-        attachLogFieldToLogs(GOVUK_SIGNIN_JOURNEY_ID, sessionCookieIds.getClientSessionId());
-
-        LOG.info("LogoutHandler processing request");
-
-        if (!session.getClientSessions().contains(sessionCookieIds.getClientSessionId())) {
-            LOG.warn("Client Session ID does not exist");
-            return logoutService.generateErrorLogoutResponse(
-                    Optional.empty(),
-                    new ErrorObject(OAuth2Error.INVALID_REQUEST_CODE, "invalid session"),
-                    input,
-                    Optional.empty(),
-                    Optional.of(session.getSessionId()));
-        }
-
-        Map<String, String> queryStringParameters = input.getQueryStringParameters();
-        if (queryStringParameters == null || queryStringParameters.isEmpty()) {
-            LOG.info("Deleting session and returning default logout as no input parameters");
-            segmentedFunctionCall("destroySessions", () -> logoutService.destroySessions(session));
-            return logoutService.generateDefaultLogoutResponse(
-                    state, input, Optional.empty(), Optional.of(session.getSessionId()));
+            state = Optional.ofNullable(queryStringParameters.get("state"));
         }
 
         Optional<String> idTokenHint =
                 Optional.ofNullable(queryStringParameters.get("id_token_hint"));
+        Optional<String> audience;
         Optional<String> postLogoutRedirectUri =
                 Optional.ofNullable(queryStringParameters.get("post_logout_redirect_uri"));
-        Optional<String> audience = Optional.empty();
-
         if (idTokenHint.isPresent()) {
             LOG.info("ID token hint is present");
-            if (!doesIDTokenExistInSession(idTokenHint.get(), session)) {
-                LOG.warn("ID token does not exist");
-                return unableToValidateIdTokenHint(input, session);
-            }
```

Comment:
Why has this been moved? I feel it makes senses to check the token exists before checking its signature

Reply:
Was confused as this was labelled as refactoring when it changed functionality. Discussed this on our call.

Reply:
We check the existence first - its the parsing thats after the validation.

Reply:
Discussed on call- id token hint is passed in in the query params and so this is checking that the token is in the session.